hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictorImpl.java

```diff
@@ -204,4 +221,208 @@ protected long getEvictionCriteriaValue(Record record, EvictionPolicy evictionPo
         return value;
     }
 
+
+    @Override
+    public boolean checkEvictionPossible(RecordStore recordStore) {
+        if (recordStore.size() < 1) {
+            return false;
+        }
+
+        String mapName = recordStore.getName();
+        int partitionId = recordStore.getPartitionId();
+
+        MapContainer mapContainer = recordStore.getMapContainer();
+        MaxSizeConfig maxSizeConfig = mapContainer.getMapConfig().getMaxSizeConfig();
+        MaxSizeConfig.MaxSizePolicy maxSizePolicy = maxSizeConfig.getMaxSizePolicy();
+
+        boolean result;
+        switch (maxSizePolicy) {
+            case PER_NODE:
+                result = checkPerNodeEviction(mapName, maxSizeConfig);
+                break;
+            case PER_PARTITION:
+                result = checkPerPartitionEviction(mapName, maxSizeConfig, partitionId);
+                break;
+            case USED_HEAP_PERCENTAGE:
+                result = checkHeapPercentageEviction(mapName, maxSizeConfig);
+                break;
+            case USED_HEAP_SIZE:
+                result = checkHeapSizeEviction(mapName, maxSizeConfig);
+                break;
+            case FREE_HEAP_PERCENTAGE:
+                result = checkFreeHeapPercentageEviction(maxSizeConfig);
+                break;
+            case FREE_HEAP_SIZE:
+                result = checkFreeHeapSizeEviction(maxSizeConfig);
+                break;
+            default:
+                throw new IllegalArgumentException("Not an appropriate max size policy [" + maxSizePolicy + ']');
```

Comment:
Probably this should be IllegalStateException since it isn't an argument to a method.